toplevel/role/role.go

```diff
@@ -173,15 +174,15 @@ func (c config) Apply(address string, entriesBytes []byte, dryRun bool, threadPo
 		return err
 	}
 
-	desiredItems := asItems(instancesToDesiredRoles[address])
-	validateUniquenessError := vault.ValidateUniqueness(desiredItems, toplevelName)
+	desiredRoles := instancesToDesiredRoles[address]
+	validateUniquenessError := ValidateRoleUniqueness(desiredRoles, toplevelName)
```

Comment:
We can run this check earlier in the function. Move to L115 (call right after `instancesToDesiredRoles` is created)